doc/source/whatsnew/v0.21.0.txt

```diff
@@ -566,6 +566,7 @@ Categorical
 - Bug in the categorical constructor with empty values and categories causing
   the ``.categories`` to be an empty ``Float64Index`` rather than an empty
   ``Index`` with object dtype (:issue:`17248`)
+- Bug in preserving name in ``set_categories``. (:issue:`17509`)
```

Comment:
Could you make this more descriptive? Maybe

    Bug in categorical operations on Series like `Series.cat.set_categories` not preserving the original Series' name (:issue:`17509`)